src/subcommand/miner/metrics.rs

```diff
@@ -7,114 +7,126 @@ pub(crate) struct Metrics {
 }
 
 impl Metrics {
-    pub fn new() -> Self {
+    pub(crate) fn new() -> Self {
         Self {
             hashes: AtomicU64::new(0),
             shares: AtomicU64::new(0),
             started: Instant::now(),
         }
     }
 
-    pub fn add_hashes(&self, hashes: u64) {
+    pub(crate) fn add_hashes(&self, hashes: u64) {
         self.hashes.fetch_add(hashes, Ordering::Relaxed);
     }
 
-    pub fn add_share(&self) {
+    pub(crate) fn add_share(&self) {
         self.shares.fetch_add(1, Ordering::Relaxed);
     }
 
-    pub fn total_hashes(&self) -> u64 {
+    pub(crate) fn total_hashes(&self) -> u64 {
         self.hashes.load(Ordering::Relaxed)
     }
 
-    pub fn total_shares(&self) -> u64 {
+    pub(crate) fn total_shares(&self) -> u64 {
         self.shares.load(Ordering::Relaxed)
     }
 
-    pub fn uptime(&self) -> Duration {
+    pub(crate) fn uptime(&self) -> Duration {
         self.started.elapsed()
     }
+
+    pub(crate) fn avg_hashrate(&self) -> f64 {
+        let hashes = self.total_hashes() as f64;
+        let secs = self.uptime().as_secs_f64();
+        if secs > 0.0 { hashes / secs } else { 0.0 }
+    }
+}
+
+impl StatusLine for Metrics {
+    fn status_line(&self) -> String {
+        format!(
+            "hashrate={}H/s  shares={}  uptime={}s",
+            ckpool::HashRate(self.avg_hashrate()),
+            self.total_shares(),
+            self.uptime().as_secs()
+        )
+    }
 }
```

Comment:
Responsive hashrate display step 4: Update `status_line()` to use the smoothed rate:
```suggestion
impl StatusLine for Metrics {
    fn status_line(&self) -> String {
        format!(
            "hashrate={}H/s  shares={}  uptime={}s",
            ckpool::HashRate(self.current_hashrate()),
            self.total_shares(),
            self.uptime().as_secs()
        )
    }
}
```